asdf_astropy/converters/unit/tests/test_quantity.py

```diff
@@ -65,3 +65,72 @@ def test_read_array_value():
     with asdf.open(buff) as af:
         assert_array_equal(af["quantity"].value, np.array([1.0, 2.0, 3.0, 4.0]))
         assert af["quantity"].unit.is_equivalent(units.km)
+
+
+def test_memmap(tmp_path):
+    """
+    Test that memmap (copy_arrays=False) works with quantities.
+
+    Unfortunately, this is not a simple `isinstance(obj, np.memmap)`
```

Comment:
In a way I guess it is OK to test just on the actual effect -- certainly, for `np.memmap`, something can be an instance without an actual file backing it.